cmd/lakectl/cmd/plugin_list.go

```diff
@@ -0,0 +1,143 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"slices"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const pluginListTemplate = `The following lakectl-compatible plugins are available:
+
+{{- range .Plugins }}
+  {{- if or .Verbose .Warning }}
+  {{ .Path }}
+  {{- else }}
+  {{ .Name }}
+  {{- end }}
+  {{- if .Warning }}
+    - {{ "warning:" | yellow }} {{ .Warning }}
+  {{- end }}
+{{- end }}
+`
+
+type pluginInfo struct {
+	Name    string
+	Path    string
+	Warning string
+	Verbose bool
+}
+
+var pluginListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List available lakectl plugins",
+	Long:  `Scans the PATH for executables named "lakectl-*" and lists the detected plugins.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		verbose, _ := cmd.Flags().GetBool("verbose")
+		envPathDirs := filepath.SplitList(os.Getenv("PATH"))
+
+		// Filter out duplicates from envPathDirs
+		var pathDirs []string
+		seen := make(map[string]struct{})
+		for _, dir := range envPathDirs {
+			if dir == "" {
+				// Skip empty entries in PATH
+				continue
+			}
+			if _, ok := seen[dir]; !ok {
+				seen[dir] = struct{}{}
+				pathDirs = append(pathDirs, dir)
+			}
+		}
+
+		// Iterate through pathDirs to build the final list with correct overshadowing
+		pluginsOutput := make([]pluginInfo, 0)
+		foundEffectivePlugins := make(map[string]string) // plugin name -> path of chosen executable
+
+		for _, dir := range pathDirs {
+			files, err := os.ReadDir(dir)
+			if err != nil {
```

Comment:
Check for the appropriate errors. Otherwise some errors will be impossible to diagnose. I suggest copying three defined behaviour of PATH in POSIX, or in MacIS, or in Linux (I suspect they're all the same).

Reply:
I prefer to show none, or show all. This is not the tool to understand whats wrong with your path, in case folder no longer exists and we fail to read it.
Will print warning with the error for all the cases the code silently ignores when verbose is turned on.

Reply:
The tool for understanding what is wrong with my path is the `which(1)` command.  BUT it only works if our program does exactly the same thing as the shell does.  If our program does something different _then `which` is no good_.

So either error reporting has to be really good, or this loop has to go, or we need to copy the essence of the parallel code in `git` or something.

Reply:
Popular shells do not report an error when there is a non-exists directory in the path, for all other cases I printed a warning while listing the folders.

Is there other errors I didn't print or you think we need to print it always?

Reply:
New code is fine, thanks!